nautobot/core/templatetags/buttons.py

```diff
@@ -124,6 +126,140 @@ def job_import_url(content_type):
     return import_url
 
 
+def render_tag_attrs(attrs_dict):
+    """Converts tag attributes from a dictionary to a string format suitable for HTML rendering."""
+    return format_html_join(" ", '{}="{}"', list(attrs_dict.items()))
+
+
+@register.inclusion_tag("buttons/consolidated_bulk_action_buttons.html")
+def consolidate_bulk_action_buttons(request, content_type, perms, bulk_edit_url, bulk_delete_url, permissions):
+    """
+    Generates a list of action buttons for bulk operations (edit, update group assignment, delete) based on the
+    provided permissions and URLs.
+
+    Args:
+        request (HttpRequest): The HTTP request object.
+        content_type (ContentType): The content type for the model being acted upon.
+        perms (dict): A dictionary of permissions for the current user.
+        bulk_edit_url (str): The URL for the bulk edit action.
+        bulk_delete_url (str): The URL for the bulk delete action.
+        permissions (dict): A dictionary of specific permissions for the actions.
+    """
+
+    bulk_action_buttons = []
+
+    render_edit_button = bool(bulk_edit_url and permissions["change"])
+    render_static_group_assign_button = bool(
+        content_type.model_class().is_dynamic_group_associable_model and perms["extras"]["add_staticgroupassociation"]
+    )
+    render_delete_button = bool(bulk_delete_url and permissions["delete"])
+    bulk_action_button_count = sum([render_edit_button, render_static_group_assign_button, render_delete_button])
+
+    if bulk_action_button_count == 0:
+        return {
+            "bulk_action_buttons": bulk_action_buttons,
+        }
+
+    params = ("?" + request.GET.urlencode()) if request.GET else ""
+
+    primary_button_fragment = child_button_fragment = """
+        <button {attrs}>
+            <span class="{icon}" aria-hidden="true"></span> {label}
+        </button>
+    """
+
+    edit_button_classes = "btn btn-sm btn-warning"
+    delete_button_classes = "btn btn-sm btn-danger"
+    static_group_button_classes = "btn btn-sm btn-primary"
+    static_group_icon = "mdi mdi-group"
```

Comment:
Why do we have a `static_group_icon` but no `edit_button_icon` or `delete_button_icon`?

Reply:
The edit button is either rendered or it isn't. No conditional icon logic needed there. The delete button icon is red so the `text-muted` class isn't needed to make it less contrasting when rendered in the dropdown